### ISSUE TITLE
[AI] Détection des stratégies et setups

### ISSUE THREAD

**Issue by antoinevanloo** [antoinevanloo/cs2-analytics#21]
## User Story
**En tant que** coach
**Je veux** que l'IA identifie automatiquement les stratégies utilisées
**Afin de** analyser rapidement les tendances de l'équipe adverse

## Description
Développer un modèle ML capable de classifier les rounds selon la stratégie/setup utilisé.

### Attaquants (T-side)
- **Default**: Répartition standard, prise d'info
- **A Execute**: Exécution rapide sur A
- **B Execute**: Exécution rapide sur B
- **A Split**: Split A depuis multiple entrées
- **B Split**: Split B
- **Mid Control**: Focus contrôle mid
- **Fast Push**: Rush sans utility
- **Fake**: Faux signaux puis rotation
- **Slow Default**: Default lent avec lurk

### Défenseurs (CT-side)
- **Standard 2-1-2**: Configuration classique
- **Stack A**: 3-4 joueurs A
- **Stack B**: 3-4 joueurs B
- **Aggressive Mid**: Contrôle agressif mid
- **Passive**: Défense passive, info
- **Retake Setup**: Setup de retake

### Features d'entrée
```python
features = [
    # Positions (normalisées par map)
    'player_positions_t0',      # Positions au freeze end
    'player_positions_t10',     # +10 secondes
    'player_positions_t20',     # +20 secondes
    
    # Timings
    'first_contact_time',
    'bomb_plant_time',
    'first_kill_location',
    
    # Utility
    'smokes_thrown_a',
    'smokes_thrown_b',
    'smokes_thrown_mid',
    'flashes_thrown',
    
    # Movement
    'group_cohesion',           # Distance entre joueurs
    'rotation_pattern',
]
```

### Modèle
```python
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import StandardScaler

class StrategyClassifier:
    def __init__(self):
        self.scaler = StandardScaler()
        self.model = RandomForestClassifier(
            n_estimators=100,
            max_depth=10,
            random_state=42
        )
    
    def predict(self, round_features: np.ndarray) -> str:
        scaled = self.scaler.transform(round_features)
        return self.model.predict(scaled)[0]
    
    def predict_proba(self, round_features: np.ndarray) -> dict:
        scaled = self.scaler.transform(round_features)
        probs = self.model.predict_proba(scaled)[0]
        return dict(zip(self.model.classes_, probs))
```

### Dataset d'entraînement
1. Collecter ~1000 rounds de démos pro
2. Labeler manuellement les stratégies
3. Augmentation des données (mirror pour maps symétriques)
4. Validation croisée

### API
```typescript
GET /analysis/:demoId/strategies

{
  rounds: [
    {
      roundNumber: 1,
      side: 'T',
      strategy: 'A_EXECUTE',
      confidence: 0.87,
      alternatives: [
        { strategy: 'A_SPLIT', confidence: 0.08 },
        { strategy: 'FAST_PUSH', confidence: 0.05 }
      ]
    }
  ],
  summary: {
    mostUsedT: 'A_EXECUTE',
    mostUsedCT: 'STANDARD',
    tendencies: { ... }
  }
}
```

## Critères d'acceptation
- [ ] Modèle entraîné avec accuracy > 85%
- [ ] API de prédiction disponible
- [ ] Visualisation des stratégies
- [ ] Temps de prédiction < 100ms/round
- [ ] Documentation du modèle

## Lié à
- Epic #3
